src/main/java/com/dynatrace/openkit/providers/DefaultThreadIDProvider.java

```diff
@@ -22,8 +22,11 @@
 public class DefaultThreadIDProvider implements ThreadIDProvider {
 
     @Override
-    public long getThreadID() {
-        return Thread.currentThread().getId();
+    public int getThreadID() {
+        long threadID64 = Thread.currentThread().getId();
+        int hash = (int)((threadID64 ^ (threadID64 >>> 32)) & 0x7fffffff );
```

Comment:
improvement suggestion: add a comment explaining that this algorithm is the way Java generates a hash code out of a long

Reply:
Can be the same comment in OpenKit C++